applications/plugins/ArticulatedSystemPlugin/src/ArticulatedSystemPlugin/ArticulatedSystemMapping.inl

```diff
@@ -523,8 +531,7 @@ void ArticulatedSystemMapping<TIn, TInRoot, TOut>::applyJT( InMatrixDeriv& out,
 
                 if(m_fromRootModel && outRoot)
                 {
-                    unsigned int indexT = (d_indexFromRoot.isSet())? d_indexFromRoot.getValue(): m_fromRootModel->getSize()-1;
-                    sofa::type::Vec<3,OutReal> posRoot = xto[indexT].getCenter();
+                    sofa::type::Vec<3,OutReal> posRoot = xfromRoot[d_indexFromRoot.getValue()].getCenter();
```

Comment:
Just highlighting the change here: `xfromRoot[indexFromRoot]` instead of `xto[indexFromRoot]`

Reply:
Yes, I could have highlighted it in my description of the PR. 

It makes more sense to me, but I would like a double check.

Reply:
Seems better for me as well! I don't understand the code behind but as d_indexFromRoot correspond and is limited to m_fromRootModel->getSize()
So using it in xto (aka m_toModel) is not relevant and could bring to a buffer overflow